src/client/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -45,6 +45,9 @@ public sealed partial class TokenCache : ITokenCacheInternal
         ITokenCacheAccessor ITokenCacheInternal.Accessor => _accessor;
         ILegacyCachePersistence ITokenCacheInternal.LegacyPersistence => LegacyCachePersistence;
 
+        bool IsAppTokenCache { get; }
```

Comment:
>bool IsAppTokenCache { get; } [](start = 8, length = 29)

private? also do we need this? Can't we simply use the one on ITokenCacheInternal?

Reply:
The pattern used here is: 

- ITokenCacheInternal is implemented as an explicit interface. `internal` interfaces can only be explicit, since the TokenCache object is public. 
- When using explicit interfaces, one needs to make an explicit cast to the interface to be able to access the methods / properties on that interface. 
- Use the trick above to overcome the explicit cast. This makes the code more readable, otherwise it would be littered with interface casting.